CT-1766 Make Business unit columns depend on parameter in Business Un…

stephenrichards opened the pull request:
…it report

The business unit columns were added to this report in order to surface cases where
the business units were responding in time, but the case was late due to other reasons.

This proved to be controversial, and so we have removed it from the report that is
generated from the stats page by making the appearance/non-appearance depend on a parameter.

misaka commented:
I'm just waiting for CT-767 to be merged so this can be re-based properly off of it.

misaka commented on app/views/cases/_case_request.html.slim:
```diff
@@ -3,9 +3,33 @@
-        = case_details.message
-      - else
-        = show_hide_message(case_details)
+    - if case_details.delivery_method == 'sent_by_email'
```
Just to say, the reason we used the `sent_by_email` wording is so that we can do: `case_details.sent_by_email?` here. :)

misaka commented on app/views/cases/_case_request.html.slim:
```diff
@@ -3,9 +3,33 @@
+      - case_details.upload_request_groups.each do |ug|
+        .case-attachments-group
+          h3.case-attachments--primary
+            = ug.date_time
```
Just remembering the issues we had elsewhere, will this be in the correct timezone?

aliuk2012 replied:
It should do because its handled in the CaseAttachmentUploadGroup model

https://github.com/ministryofjustice/correspondence_tool_staff/blob/develop/app/models/case_attachment_upload_group.rb#L17

misaka replied:
👍 I guess it's easily confirmed seeing as we're currently in BST.

stephenrichards replied:
Yes - the upload group takes care of putting it in the right timezone

misaka commented on app/models/case.rb:
```diff
@@ -214,8 +214,12 @@ def self.search(term)
 
-  def upload_groups
-    CaseAttachmentUploadGroupCollection.new(self, attachments)
+  def upload_response_groups
```
We should have tests for this and `upload_request_groups` below.

stephenrichards committed:
CT-1766 Make Business unit columns depend on parameter in Business Unit report

The business unit columns were added to this report in order to surface cases where
the business units were responding in time, but the case was late due to other reasons.

This proved to be controversial, and so we have removed it from the report that is
generated from the stats page by making the appearance/non-appearance depend on a parameter.
